File: pmcjs/samples/oleado.js

use('Perl');
Perl.Use('Win32::OLE');
Perl.Methods['Win32::OLE'].Option('CP', Perl.Constants['Win32::OLE'].CP_UTF8);
Perl.Use('Win32::OLE::Variant');
var Variant = Perl.Functions['Win32::OLE::Variant'];

var db = new Perl('Win32::OLE', 'ADODB.Connection');
db.Provider = 'Microsoft.Jet.OLEDB.4.0';
db.Mode = 1;
db.ConnectionString = 'adotest.mdb';
db.Open;

var rs = db.Execute('select * from testtbl');
var n = rs.Fields.Count;
for (var i = 0; i < n; ++i) {
  print('fld ' + i + ': ' + rs.Fields.Item(i).Name);
}
while (!rs.EOF) {
  print();
  var rec = rs.Fields;
  for (var i = 0; i < n; ++i) {
    var val = rec.Item(i).Value;
    if (Perl.IsPerl(val)) {
      var v = Variant.Variant(Variant.VT_DATE(), val);
      if (v) {
	val = v.Date('yyyy/MM/dd') + ' ' + v.Time('HH:mm:ss');
      } else {
        val = null;
      }
    }
    print('value: ' + val);
  }
  rs.MoveNext;
}
db.Close;
print('done');

File: pmcjs/scripts/oleado.pl

use Win32::OLE;
use Win32::OLE::Variant;
my $db = Win32::OLE->new('ADODB.Connection');
$db->{Provider} = 'Microsoft.Jet.OLEDB.4.0';
$db->{Mode} = 1;
$db->{ConnectionString} = 'adotest.mdb';
$db->Open;

my $rs = $db->Execute('select * from testtbl');
my $n = $rs->{Fields}->{Count};
for (my $i = 0; $i < $n; ++$i) {
  print('fld $i: ', $rs->{Fields}->Item($i)->{Name}, "\n");
}
while (!$rs->{EOF}) {
  print("\n");
  my $rec = $rs->{Fields};
  for (my $i = 0; $i < $n; ++$i) {
    my $v = $rec->Item($i)->{Value};
    if (ref($v)) {
      my $vd = Variant(VT_DATE, $v);
      $v = $vd->Date('yyyy/MM/dd') . ' ' . $vd->Time('HH:mm:ss');
    }
    print('value: ', $v, "\n");
  }
  $rs->MoveNext;
}
